drivers/vultr/vultr.go

```diff
@@ -0,0 +1,517 @@
+package vultr
+
+import (
+	"bytes"
+	"fmt"
+	"io/ioutil"
+	"path/filepath"
+	"strconv"
+	"strings"
+	"text/template"
+	"time"
+
+	"github.com/ChimeraCoder/tokenbucket"
+	vultr "github.com/JamesClonk/vultr/lib"
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/drivers"
+	"github.com/docker/machine/log"
+	"github.com/docker/machine/ssh"
+	"github.com/docker/machine/state"
+)
+
+type Driver struct {
+	APIKey            string
+	MachineID         string
+	MachineName       string
+	IPAddress         string
+	PrivateIP         string
+	OSID              int
+	RegionID          int
+	PlanID            int
+	SSHKeyID          string
+	SSHUser           string
+	SSHPort           int
+	IPv6              bool
+	Backups           bool
+	PrivateNetworking bool
+	CaCertPath        string
+	PrivateKeyPath    string
+	DriverKeyPath     string
+	SwarmMaster       bool
+	SwarmHost         string
+	SwarmDiscovery    string
+	storePath         string
+	scriptID          int
+	bucket            *tokenbucket.Bucket
+}
+
+func init() {
+	drivers.Register("vultr", &drivers.RegisteredDriver{
+		New:            NewDriver,
+		GetCreateFlags: GetCreateFlags,
+	})
+}
+
+// GetCreateFlags registers the flags this driver adds to "docker hosts create"
+func GetCreateFlags() []cli.Flag {
+	return []cli.Flag{
+		cli.StringFlag{
+			EnvVar: "VULTR_API_KEY",
+			Name:   "vultr-api-key",
+			Usage:  "Vultr API key",
+		},
+		cli.IntFlag{
+			EnvVar: "VULTR_OS",
+			Name:   "vultr-os-id",
+			Usage:  "Vultr operating system ID. Default: 159 (RancherOS)",
+			Value:  159,
```

Comment:
Not that I disagree, but...
RancherOS can be configured to look like Ubuntu :)
It's just a matter of what console container to use. Currently, RancherOS
does use ubuntu-console by default in cloud deployments.

Just my 2 cents.

On Tue, Jun 23, 2015, 23:41 Dave Henderson <notifications@github.com> wrote:

> In drivers/vultr/vultr.go
> <https://github.com/docker/machine/pull/958#discussion_r33076225>:
>
> > +	})
> > +}
> > +
> > +// GetCreateFlags registers the flags this driver adds to "docker hosts create"
> > +func GetCreateFlags() []cli.Flag {
> > +	return []cli.Flag{
> > +		cli.StringFlag{
> > +			EnvVar: "VULTR_API_KEY",
> > +			Name:   "vultr-api-key",
> > +			Usage:  "Vultr API key",
> > +		},
> > +		cli.IntFlag{
> > +			EnvVar: "VULTR_OS",
> > +			Name:   "vultr-os-id",
> > +			Usage:  "Vultr operating system ID. Default: 159 (RancherOS)",
> > +			Value:  159,
>
> So... While RancherOS is definitely cool, and is considered "supported"
> from docker-machine's point of view, the default for remote drivers (like
> Vultr) is Ubuntu (see
> https://github.com/docker/machine/blob/master/docs/index.md#specify-a-base-operating-systems).
> So IMO you should change the default to Ubuntu (probably 14.04 - i.e. 160),
> just to keep in line with what everyone expects and assumes.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/docker/machine/pull/958/files#r33076225>.
>